src/meta/processors/admin/ListClusterInfoProcessor.cpp

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/admin/ListClusterInfoProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void ListClusterInfoProcessor::process(const cpp2::ListClusterInfoReq& req) {
+    UNUSED(req);
+    auto* store = dynamic_cast<kvstore::NebulaStore*>(kvstore_);
+    if (store == nullptr) {
+        onFinished();
+        return;
+    }
+
+    if (!store->isLeader(kDefaultSpaceId, kDefaultPartId)) {
+        handleErrorCode(cpp2::ErrorCode::E_LEADER_CHANGED);
+        onFinished();
+        return;
+    }
+
+    const auto& prefix = MetaServiceUtils::hostPrefix();
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "prefix failed:" << ret;
+        handleErrorCode(cpp2::ErrorCode::E_LIST_CLUSTER_FAILURE);
+        onFinished();
+        return;
+    }
+
+    std::vector<nebula::cpp2::NodeInfo> storages;
+    while (iter->valid()) {
+        auto host = MetaServiceUtils::parseHostKey(iter->key());
+        HostInfo info = HostInfo::decode(iter->val());
+
+        if (info.role_ != cpp2::HostRole::STORAGE) {
+            iter->next();
+            continue;
+        }
+
+        auto status = client_->listClusterInfo(host).get();
```

Comment:
For now, all storage need to be alive when backup, am I right?
There is a case that, there would be some hosts which is dead for a long time, and data on it has been transferred. I have not figured out a way of how to get all storage need to backup.

Reply:
Yes, maybe we need to specify that in the documentation.